### ISSUE TITLE
2.7.1 Destroyed the Ability to Pass in an `Api` Instance to Objects

### ISSUE THREAD

**Issue by chrisguitarguy** [facebook/facebook-php-business-sdk#243]
Specifically all the association methods on objects (like `ProductFeed::getUploads`) were switched from [using `getManyByConnection`](https://github.com/facebook/facebook-php-ads-sdk/blob/2.6.0/src/FacebookAds/Object/ProductFeed.php#L53) to [creating an `ApiRequest` and child object directly](https://github.com/facebook/facebook-php-ads-sdk/blob/2.7.1/src/FacebookAds/Object/ProductFeed.php#L90).

`getManyByConnection` provides the `FacebookAds\Api` instance on the parent to the child. But the new method relies on setting up global state via `Api::instance`.

In short, every time an object does `new ChildObjectType(...)` it should pass the `Api` instance its given to the child. Right now it doesn't and it breaks things. A lot of things.

**Comment by chrisguitarguy** [comment]
Seems like a workaround is to always call `Api::init` in some way so `new WhateverObject()` that's done to create a prototype object for `ApiRequest` doesn't fail. `ApiRequest` and `Cursor` will call `setApi` on any classes that extend `AbstractCrubObject` to make sure they get the correct instance.

Maybe a fix here to document that a call to `Api::init` is required at some point in your application lifecycle, even if managing sessions manually by passing in `Api` objects elsewhere.

**Comment by jingping2015** [comment]
Hi, it seems like this issue has been open for more than a few months now. We have made many changes in the SDK since then, including releasing a new version. If you can repro your specific issue with the latest version of the SDK, please create a new issue or comment here with further details.
